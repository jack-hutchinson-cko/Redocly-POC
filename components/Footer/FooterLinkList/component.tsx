import React, { useState } from 'react';
import { Chevron } from '../../svg';
import {
	Container,
	SolutionsLinksContainer,
	LinksContainer,
	ColMainWrapper,
	ColWrapper,
	FooterTitle,
	FooterLink,
	LinkText,
	StyledIcon,
} from './styles';

const FooterLinkList = ({ links = [] }) => {
	const [showSolutions, setSolutions] = useState<boolean>(false);
	const [showCompany, setCompany] = useState<boolean>(false);
	const [showInsights, setInsights] = useState<boolean>(false);
	const [showResources, setResources] = useState<boolean>(false);

	const setFuncs = [
		() => setCompany(!showCompany),
		() => setInsights(!showInsights),
		() => setResources(!showResources),
	];
	const showStatements = [showCompany, showInsights, showResources];

	return (
		<Container>
			{links?.map((link, i) => {
				if (i === 0) {
					return (
						<SolutionsLinksContainer key={i} onClick={() => setSolutions(!showSolutions)}>
							<FooterTitle>
								{link.primary.title[0].text}
								<StyledIcon>
									<Chevron />
								</StyledIcon>
							</FooterTitle>
							<ColMainWrapper>
								<ColWrapper isDisplayed={showSolutions}>
									{link.items.slice(0, link.items.length / 2 + 1).map((item, i) => (
										<FooterLink href={item.link_href[0].text} labelHere={item.link_name[0].text === 'Careers'} key={i}>
											<LinkText> {item.link_name[0].text} </LinkText>
										</FooterLink>
									))}
								</ColWrapper>

								<ColWrapper isDisplayed={showSolutions}>
									{link.items.slice(link.items.length / 2 + 1, link.items.length).map((item, i) => (
										<FooterLink href={item.link_href[0].text} labelHere={item.link_name[0].text === 'Careers'} key={i}>
											<LinkText> {item.link_name[0].text} </LinkText>
										</FooterLink>
									))}
								</ColWrapper>
							</ColMainWrapper>
						</SolutionsLinksContainer>
					);
				} else {
					return (
						<LinksContainer key={i} onClick={setFuncs[i - 1]}>
							<FooterTitle>
								{link.primary.title[0].text}
								<StyledIcon>
									<Chevron />
								</StyledIcon>
							</FooterTitle>
							<ColWrapper isDisplayed={showStatements[i - 1]}>
								{link.items.map((item, i) => (
									<FooterLink href={item.link_href[0].text} labelHere={item.link_name[0].text === 'Careers'} key={i}>
										<LinkText> {item.link_name[0].text} </LinkText>
									</FooterLink>
								))}
							</ColWrapper>
						</LinksContainer>
					);
				}
			})}
		</Container>
	);
};

export default FooterLinkList;
