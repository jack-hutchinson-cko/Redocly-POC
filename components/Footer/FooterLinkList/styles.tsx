import styled from 'styled-components';
import { device } from '../../css/tokens';

export const Container = styled.div`
	display: flex;
	justify-content: space-between;

	${device.toTablet} {
		flex-direction: column;
	}
`;

export const SolutionsLinksContainer = styled.div`
	width: 32%;
	margin-top: 20px;

	@media only screen and (min-width: 768px) and (max-width: 1080px) {
		margin-top: 40px;
		padding: 0 0 20px 0;
		width: 36%;
	}

	${device.toTablet} {
		border-bottom: 1px solid var(--border);
		margin-top: 40px;
		padding: 0 0 20px 0;
		width: auto;
	}
`;

export const LinksContainer = styled.div`
	margin-top: 20px;

	${device.toTablet} {
		border-bottom: 1px solid var(--border);
		padding: 0 0 20px 0;
	}
`;

export const ColMainWrapper = styled.div`
	display: flex;
	justify-content: space-between;

	${device.toTablet} {
		flex-direction: column;
	}
`;

export const ColWrapper = styled.div`
	margin-top: 16px;
	display: flex;
	flex-direction: column;

	${device.toTablet} {
		display: ${(props: { isDisplayed: boolean }) => (props.isDisplayed ? 'flex' : 'none')};
		transition: all 0.5s;
		&.solutions-second-col {
			margin-top: 0;
		}
	}
`;

export const FooterTitle = styled.div`
	font-style: normal;
	font-weight: 600;
	font-size: 12px;
	line-height: 16px;
	letter-spacing: 0.11em;
	text-transform: uppercase;
	color: white;
	display: flex;
	justify-content: space-between;
	-webkit-font-smoothing: antialiased;
`;

export const FooterLink = styled.a`
	font-style: normal;
	font-weight: normal;
	font-size: 14px;
	line-height: 40px;
	text-decoration: none;
	color: var(--S20);
	-webkit-font-smoothing: antialiased;

	:hover div {
		border-bottom: 1px solid var(--S20);
	}

	${(props: { labelHere: boolean }) =>
		props.labelHere
			? `&::after {
        content:  "WE'RE HIRING";
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        line-height: 16px;
        text-transform: uppercase;
        color: var(--font-color);
        background: #2BD4DB;
        border-radius: 4px;
        margin-left: 8px;
        padding: 4px;
        text-decoration: none;
      }`
			: ''};

	&:hover:after {
		text-decoration: none;
	}

	${device.toMobileXL} {
		line-height: 32px;
	}
`;

export const LinkText = styled.div`
	border-bottom: 1px solid transparent;
	display: inline;
	padding-bottom: 4px;
	transition: border-bottom-color 0.2s ease;
`;

export const StyledIcon = styled.div`
	display: none;

	${device.toTablet} {
		display: flex;
		align-items: center;
	}
`;
