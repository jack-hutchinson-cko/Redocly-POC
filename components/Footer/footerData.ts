export const dataFooter = {
	body: [
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Solution overview',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Unified payments platform',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/unified-api',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'International Coverage',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/international-coverage',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Data and Reporting',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/data-and-reporting',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Integrations',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/integration',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Fraud protection',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/solutions/fraud',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Payment methods',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/payment-methods',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Pricing',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/pricing',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Partners',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/partners',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Solution',
						spans: [],
					},
				],
			},
		},
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'About us',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/about-us',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Careers',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/careers',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Newsroom',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/newsroom',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Contact us',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/contact-us',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Become a partner',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/partners/join',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Company',
						spans: [],
					},
				],
			},
		},
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Blog',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/blog',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Customer stories',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: '/case-studies',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Insights',
						spans: [],
					},
				],
			},
		},
		{
			slice_type: 'links_section',
			slice_label: null,
			items: [
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Documentation',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: 'https://docs.checkout.com/',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'API Reference',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: 'https://api-reference.checkout.com/',
							spans: [],
						},
					],
				},
				{
					link_name: [
						{
							type: 'paragraph',
							text: 'Support',
							spans: [],
						},
					],
					link_href: [
						{
							type: 'paragraph',
							text: 'mailto:support@checkout.com',
							spans: [],
						},
					],
				},
			],
			primary: {
				title: [
					{
						type: 'heading2',
						text: 'Resources',
						spans: [],
					},
				],
			},
		},
	],
	title: [{ title: 'mock data' }],
};
