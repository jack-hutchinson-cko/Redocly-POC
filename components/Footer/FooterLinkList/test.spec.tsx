import { shallow } from 'enzyme';
import React from 'react';
import FooterLinkList from './component';
import { footerData } from '../footerData';

describe('FooterLinkList component', () => {
	test.each`
		links
		${null}
		${footerData.body}
	`('component matches snapshot', ({ links }) => {
		expect.assertions(1);

		const wrapper = shallow(<FooterLinkList links={links} />);

		expect(wrapper).toMatchSnapshot();
	});
});
