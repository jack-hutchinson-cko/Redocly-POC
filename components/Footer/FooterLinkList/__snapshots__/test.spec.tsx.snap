// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterLinkList component component matches snapshot 1`] = `<styled.div />`;

exports[`FooterLinkList component component matches snapshot 2`] = `
<styled.div>
  <styled.div
    key="0"
  >
    <styled.p>
      Solution
    </styled.p>
    <styled.div>
      <styled.div>
        <styled.a
          href="/solutions"
          key="0"
          labelHere={false}
        >
          <styled.div>
             
            Solution overview
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/solutions/unified-api"
          key="1"
          labelHere={false}
        >
          <styled.div>
             
            Unified payments platform
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/solutions/international-coverage"
          key="2"
          labelHere={false}
        >
          <styled.div>
             
            International Coverage
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/solutions/data-and-reporting"
          key="3"
          labelHere={false}
        >
          <styled.div>
             
            Data and Reporting
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/solutions/integration"
          key="4"
          labelHere={false}
        >
          <styled.div>
             
            Integrations
             
          </styled.div>
        </styled.a>
      </styled.div>
      <styled.div
        className="solutions-second-col"
      >
        <styled.a
          href="/solutions/fraud"
          key="0"
          labelHere={false}
        >
          <styled.div>
             
            Fraud protection
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/payment-methods"
          key="1"
          labelHere={false}
        >
          <styled.div>
             
            Payment methods
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/pricing"
          key="2"
          labelHere={false}
        >
          <styled.div>
             
            Pricing
             
          </styled.div>
        </styled.a>
        <styled.a
          href="/partners"
          key="3"
          labelHere={false}
        >
          <styled.div>
             
            Partners
             
          </styled.div>
        </styled.a>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    key="1"
  >
    <styled.p>
      Company
    </styled.p>
    <styled.div>
      <styled.a
        href="/about-us"
        key="0"
        labelHere={false}
      >
        <styled.div>
           
          About us
           
        </styled.div>
      </styled.a>
      <styled.a
        href="/careers"
        key="1"
        labelHere={true}
      >
        <styled.div>
           
          Careers
           
        </styled.div>
      </styled.a>
      <styled.a
        href="/newsroom"
        key="2"
        labelHere={false}
      >
        <styled.div>
           
          Newsroom
           
        </styled.div>
      </styled.a>
      <styled.a
        href="/contact-us"
        key="3"
        labelHere={false}
      >
        <styled.div>
           
          Contact us
           
        </styled.div>
      </styled.a>
      <styled.a
        href="/partners/join"
        key="4"
        labelHere={false}
      >
        <styled.div>
           
          Become a partner
           
        </styled.div>
      </styled.a>
    </styled.div>
  </styled.div>
  <styled.div
    key="2"
  >
    <styled.p>
      Insights
    </styled.p>
    <styled.div>
      <styled.a
        href="/blog"
        key="0"
        labelHere={false}
      >
        <styled.div>
           
          Blog
           
        </styled.div>
      </styled.a>
      <styled.a
        href="/case-studies"
        key="1"
        labelHere={false}
      >
        <styled.div>
           
          Customer stories
           
        </styled.div>
      </styled.a>
    </styled.div>
  </styled.div>
  <styled.div
    key="3"
  >
    <styled.p>
      Resources
    </styled.p>
    <styled.div>
      <styled.a
        href="https://docs.checkout.com/"
        key="0"
        labelHere={false}
      >
        <styled.div>
           
          Documentation
           
        </styled.div>
      </styled.a>
      <styled.a
        href="https://api-reference.checkout.com/"
        key="1"
        labelHere={false}
      >
        <styled.div>
           
          API Reference
           
        </styled.div>
      </styled.a>
      <styled.a
        href="mailto:support@checkout.com"
        key="2"
        labelHere={false}
      >
        <styled.div>
           
          Support
           
        </styled.div>
      </styled.a>
    </styled.div>
  </styled.div>
</styled.div>
`;
